pkg/test/fake/fake.go

```diff
@@ -0,0 +1,43 @@
+package fake
+
+import (
+	"reflect"
+
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	hivecontractsv1alpha1 "github.com/openshift/hive/apis/hivecontracts/v1alpha1"
+	hiveintv1alpha1 "github.com/openshift/hive/apis/hiveinternal/v1alpha1"
+	scheme "github.com/openshift/hive/pkg/util/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func NewFakeClientBuilder() *fake.ClientBuilder {
```

Comment:
Really need some comments on this thing. I understand it because we talked about it, but our posterity would have no clue what's going on here.